File: app/src/main/java/com/legue/axel/lolsummonertool/database/viewmodel/SpellViewModel.java
package com.legue.axel.lolsummonertool.database.viewmodel;

import android.app.Application;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.annotation.NonNull;

import com.legue.axel.lolsummonertool.database.SummonerToolDatabase;
import com.legue.axel.lolsummonertool.database.model.champion.SpellImage;

public class SpellViewModel extends AndroidViewModel {

    private SummonerToolDatabase database;


    public SpellViewModel(@NonNull Application application) {
        super(application);
        database = SummonerToolDatabase.getInstance(this.getApplication());
    }

    public LiveData<SpellImage> getSpellImage(String spellName) {
        return database.spellImageDao().getSpellImageBySpellId(spellName);
    }
}

File: app/src/main/java/com/legue/axel/lolsummonertool/database/viewmodel/SpellViewModel.kt
package com.legue.axel.lolsummonertool.database.viewmodel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData

import com.legue.axel.lolsummonertool.database.SummonerToolDatabase
import com.legue.axel.lolsummonertool.database.model.champion.SpellImage

class SpellViewModel(application: Application) : AndroidViewModel(application) {

    private val database: SummonerToolDatabase = SummonerToolDatabase.getInstance(this.getApplication())


    fun getSpellImage(spellName: String): LiveData<SpellImage> {
        return database.spellImageDao().getSpellImageBySpellId(spellName)
    }
}

